tests/gdb-tests/tests/test_gdblib_parameter.py

```diff
@@ -6,31 +6,57 @@
 
 @pytest.mark.parametrize(
     "params",
-    (("int", 123, "123"), ("bool", True, "on"), ("string", "some-string-val", "some-string-val")),
+    (
+        ("int", 123, "123", {}),
+        ("bool", True, "on", {}),
+        ("bool", False, "off", {}),
+        ("string", "some-string-val", "some-string-val", {}),
+        ("auto-bool", None, "auto", {"param_class": gdb.PARAM_AUTO_BOOLEAN}),
+        ("unlimited-uint", 0, "unlimited", {"param_class": gdb.PARAM_UINTEGER}),
+        ("unlimited-int", 0, "unlimited", {"param_class": gdb.PARAM_INTEGER}),
+        ("unlimited-zuint", -1, "unlimited", {"param_class": gdb.PARAM_ZUINTEGER_UNLIMITED}),
+        (
+            "enum",
+            "enum1",
+            "enum1",
+            {"param_class": gdb.PARAM_ENUM, "enum_sequence": ["enum1", "enum2", "enum3"]},
+        ),
+    ),
 )
 def test_gdb_parameter_default_value_works(start_binary, params):
-    name_suffix, default_value, displayed_value = params
+    name_suffix, default_value, displayed_value, optional_kwargs = params
 
     param_name = f"test-param-{name_suffix}"
+    help_docstring = f"Help docstring for {param_name}"
+    optional_kwargs["help_docstring"] = help_docstring
 
-    param = pwndbg.gdblib.config.add_param(param_name, default_value, "some show string")
+    param = pwndbg.gdblib.config.add_param(
+        param_name, default_value, "some show string", **optional_kwargs
+    )
 
     # Initialize and register param in GDB as if it would be done by gdblib.config.init_params
     pwndbg.gdblib.config_mod.Parameter(param)
 
     out = gdb.execute(f"show {param_name}", to_string=True)
-    assert out in (
-        f"""The current value of '{param_name}' is "{displayed_value}".\n""",  # GDB 12.x
-        f"Show some show string {displayed_value}\n",  # GDB 9.x
-    )
-    assert gdb.parameter(param_name) == default_value
+    assert out == f"The current value of {param_name!r} is {displayed_value!r}\n"
+    if (
+        optional_kwargs.get("param_class") in (gdb.PARAM_UINTEGER, gdb.PARAM_INTEGER)
+        and default_value == 0
+    ):
+        # Note: This is really weird, according to GDB docs, 0 should mean "unlimited" for gdb.PARAM_UINTEGER and gdb.PARAM_INTEGER, but somehow GDB sets the value to `None` actually :/
+        # And hilarious thing is that GDB won't let you set the default value to `None` when you construct the `gdb.Parameter` object with `gdb.PARAM_UINTEGER` or `gdb.PARAM_INTEGER` lol
+        # Maybe it's a bug of GDB?
```

Comment:
Let's consider reporting this upstream if we can.

Reply:
Ok, I sent it to the mailing list of GDB now.

Reply:
Link the discussion in a comment reply to this PR pls

Reply:
https://sourceware.org/pipermail/gdb/2022-October/050398.html

Reply:
> https://sourceware.org/pipermail/gdb/2022-October/050398.html

Update: https://sourceware.org/pipermail/gdb/2022-October/050426.html
If I didn't misunderstand something, seems like the value in Python to represent `unlimited` will only be `None` instead of some integers in the future.
So we'll probably need to update the code here, the tests' params, and `_value_to_gdb_native()` to fit the new API in the future.